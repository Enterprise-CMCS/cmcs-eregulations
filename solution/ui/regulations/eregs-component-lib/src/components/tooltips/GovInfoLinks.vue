<template>
    <div class="gov-info-links-container">
        <div class="gov-info-links">
            <SimpleSpinner v-if="loading" size="medium" />
            <div v-else-if="govInfoLinks.length === 0" class="no-results">
                No results found.
            </div>
            <div v-else class="links-container">
                <a
                    v-for="(yearObj, index) in govInfoLinks"
                    :key="index"
                    :href="yearObj.link"
                    class="external"
                    target="_blank"
                    rel="noopener noreferrer"
                >{{ yearObj.year }}</a>
            </div>
        </div>
        <div class="gov-info-source">
            Source: CFR Annual Edition from
            <a
                href="https://www.govinfo.gov/app/collection/cfr"
                class="external"
                target="_blank"
                rel="noopener noreferrer"
            >
                GovInfo</a>
            (1996â€“Present)
        </div>
    </div>
</template>

<script>
import { getGovInfoLinks } from "utilities/api";
import SimpleSpinner from "../SimpleSpinner.vue";

export default {
    name: "GovInfoLinks",

    components: {
        SimpleSpinner,
    },

    props: {
        apiUrl: {
            type: String,
            required: true,
        },
        title: {
            type: String,
            required: true,
        },
        part: {
            type: String,
            required: true,
        },
        section: {
            type: String,
            required: true,
        },
    },

    created() {
        getGovInfoLinks({
            apiUrl: this.apiUrl,
            filterParams: {
                title: this.title,
                part: this.part,
                section: this.section,
            },
        })
            .then((response) => {
                this.govInfoLinks = response.sort((a, b) => b.year - a.year);
            })
            .catch((error) => {
                console.error("Error", error);
                this.govInfoLinks = [];
            })
            .finally(() => {
                this.loading = false;
            });
    },

    data() {
        return {
            govInfoLinks: [],
            loading: true,
        };
    },
};
</script>
