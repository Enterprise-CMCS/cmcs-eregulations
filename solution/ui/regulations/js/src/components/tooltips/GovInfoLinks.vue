<template>
    <div class="gov-info-links-container">
        <div class="gov-info-links">
            <SimpleSpinner size="medium" />
        </div>
        <div class="gov-info-source">
            Section text is revised as of October 1st of each listed year. Link
            sources: <a href="https://www.govinfo.gov">GovInfo.gov</a>.
        </div>
    </div>
</template>

<script>
import SimpleSpinner from "../SimpleSpinner.vue";

export default {
    name: "GovInfoLinks",

    components: {
        SimpleSpinner,
    },

    props: {
        title: {
            type: String,
            required: true,
        },
        part: {
            type: String,
            required: true,
        },
        section: {
            type: String,
            required: true,
        },
    },

    beforeCreate() {},

    created() {},

    beforeMount() {},

    mounted() {
        console.log("this.title", this.title);
        console.log("this.part", this.part);
        console.log("this.section", this.section);
    },

    beforeUpdate() {},

    updated() {},

    beforeDestroy() {},

    destroyed() {},

    data() {
        return {
            govInfoLinks: [],
        };
    },

    computed: {
        computedProp() {
            return this.dataProp.toUpperCase();
        },
    },

    methods: {
        methodName() {
            console.log("method has been invoked");
        },
    },
};
</script>

<style></style>
