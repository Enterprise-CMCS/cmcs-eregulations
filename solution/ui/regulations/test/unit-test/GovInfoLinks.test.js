import { render, screen } from "@testing-library/vue";
import { describe, it, expect, beforeEach, afterEach } from "vitest";
import GovInfoLinks from "eregsComponentLib/src/components/tooltips/GovInfoLinks.vue";
import flushPromises from "flush-promises";

describe("Gov Info Links", () => {
    beforeEach(() => {});
    afterEach(() => {});
    it("Populates checks if years populate", async () => {
        render(GovInfoLinks, {
            props: {
                apiUrl: "http://localhost:9000/",
                title: "42",
                part: "431",
                section: "10"
            }
        });
        await flushPromises();
        const yearLink = screen.getByText("2022");
        expect(yearLink.href).toStrictEqual(
            "https://www.govinfo.gov/content/pkg/CFR-2022-title42-vol4/pdf/CFR-2022-title42-vol4-sec431-10.pdf"
        );
        const missingYear = screen.queryByText("189829829");
        expect(missingYear).toBeFalsy();
    });
    it("Creates a snapshot of GovInfo", async () => {
        const wrapper = render(GovInfoLinks, {
            props: {
                apiUrl: "http://localhost:8000/",
                title: "42",
                part: "431",
                section: "10"
            }
        });
        await flushPromises();
        expect(wrapper).toMatchSnapshot();
    });
});
