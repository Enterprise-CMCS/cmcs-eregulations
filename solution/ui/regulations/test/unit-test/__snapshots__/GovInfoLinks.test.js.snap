// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Gov Info Links > Creates a snapshot of GovInfo 1`] = `
{
  "baseElement": <body>
    <div>
      <div
        class="gov-info-links-container"
      >
        <div
          class="gov-info-links"
        >
          <div
            class="links-container"
          >
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2022-title42-vol4/pdf/CFR-2022-title42-vol4-sec431-10.pdf"
              target="_blank"
            >
              2022
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2021-title42-vol4/pdf/CFR-2021-title42-vol4-sec431-10.pdf"
              target="_blank"
            >
              2021
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2020-title42-vol4/pdf/CFR-2020-title42-vol4-sec431-10.pdf"
              target="_blank"
            >
              2020
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2019-title42-vol4/pdf/CFR-2019-title42-vol4-sec431-10.pdf"
              target="_blank"
            >
              2019
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2018-title42-vol4/pdf/CFR-2018-title42-vol4-sec431-10.pdf"
              target="_blank"
            >
              2018
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2017-title42-vol4/pdf/CFR-2017-title42-vol4-sec431-10.pdf"
              target="_blank"
            >
              2017
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2016-title42-vol4/pdf/CFR-2016-title42-vol4-sec431-10.pdf"
              target="_blank"
            >
              2016
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2015-title42-vol4/pdf/CFR-2015-title42-vol4-sec431-10.pdf"
              target="_blank"
            >
              2015
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2014-title42-vol4/pdf/CFR-2014-title42-vol4-sec431-10.pdf"
              target="_blank"
            >
              2014
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2013-title42-vol4/pdf/CFR-2013-title42-vol4-sec431-10.pdf"
              target="_blank"
            >
              2013
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2012-title42-vol4/pdf/CFR-2012-title42-vol4-sec431-10.pdf"
              target="_blank"
            >
              2012
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2011-title42-vol4/pdf/CFR-2011-title42-vol4-sec431-10.pdf"
              target="_blank"
            >
              2011
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2010-title42-vol4/pdf/CFR-2010-title42-vol4-sec431-10.pdf"
              target="_blank"
            >
              2010
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2009-title42-vol4/pdf/CFR-2009-title42-vol4-sec431-10.pdf"
              target="_blank"
            >
              2009
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2008-title42-vol4/pdf/CFR-2008-title42-vol4-sec431-10.pdf"
              target="_blank"
            >
              2008
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2007-title42-vol4/pdf/CFR-2007-title42-vol4-sec431-10.pdf"
              target="_blank"
            >
              2007
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2006-title42-vol4/pdf/CFR-2006-title42-vol4-sec431-10.pdf"
              target="_blank"
            >
              2006
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2005-title42-vol3/pdf/CFR-2005-title42-vol3-sec431-10.pdf"
              target="_blank"
            >
              2005
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2004-title42-vol3/pdf/CFR-2004-title42-vol3-sec431-10.pdf"
              target="_blank"
            >
              2004
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2003-title42-vol3/pdf/CFR-2003-title42-vol3-sec431-10.pdf"
              target="_blank"
            >
              2003
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2002-title42-vol3/pdf/CFR-2002-title42-vol3-sec431-10.pdf"
              target="_blank"
            >
              2002
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2001-title42-vol3/pdf/CFR-2001-title42-vol3-sec431-10.pdf"
              target="_blank"
            >
              2001
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-2000-title42-vol3/pdf/CFR-2000-title42-vol3-sec431-10.pdf"
              target="_blank"
            >
              2000
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-1999-title42-vol3/pdf/CFR-1999-title42-vol3-sec431-10.pdf"
              target="_blank"
            >
              1999
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-1998-title42-vol3/pdf/CFR-1998-title42-vol3-sec431-10.pdf"
              target="_blank"
            >
              1998
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-1997-title42-vol3/pdf/CFR-1997-title42-vol3-sec431-10.pdf"
              target="_blank"
            >
              1997
            </a>
            <a
              class="external"
              href="https://www.govinfo.gov/content/pkg/CFR-1996-title42-vol3/pdf/CFR-1996-title42-vol3-sec431-10.pdf"
              target="_blank"
            >
              1996
            </a>
          </div>
        </div>
        <div
          class="gov-info-source"
        >
           Source: CFR Annual Edition from 
          <a
            class="external"
            href="https://www.govinfo.gov/app/collection/cfr"
            target="_blank"
          >
             GovInfo
          </a>
           (1996â€“Present) 
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="gov-info-links-container"
    >
      <div
        class="gov-info-links"
      >
        <div
          class="links-container"
        >
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2022-title42-vol4/pdf/CFR-2022-title42-vol4-sec431-10.pdf"
            target="_blank"
          >
            2022
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2021-title42-vol4/pdf/CFR-2021-title42-vol4-sec431-10.pdf"
            target="_blank"
          >
            2021
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2020-title42-vol4/pdf/CFR-2020-title42-vol4-sec431-10.pdf"
            target="_blank"
          >
            2020
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2019-title42-vol4/pdf/CFR-2019-title42-vol4-sec431-10.pdf"
            target="_blank"
          >
            2019
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2018-title42-vol4/pdf/CFR-2018-title42-vol4-sec431-10.pdf"
            target="_blank"
          >
            2018
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2017-title42-vol4/pdf/CFR-2017-title42-vol4-sec431-10.pdf"
            target="_blank"
          >
            2017
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2016-title42-vol4/pdf/CFR-2016-title42-vol4-sec431-10.pdf"
            target="_blank"
          >
            2016
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2015-title42-vol4/pdf/CFR-2015-title42-vol4-sec431-10.pdf"
            target="_blank"
          >
            2015
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2014-title42-vol4/pdf/CFR-2014-title42-vol4-sec431-10.pdf"
            target="_blank"
          >
            2014
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2013-title42-vol4/pdf/CFR-2013-title42-vol4-sec431-10.pdf"
            target="_blank"
          >
            2013
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2012-title42-vol4/pdf/CFR-2012-title42-vol4-sec431-10.pdf"
            target="_blank"
          >
            2012
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2011-title42-vol4/pdf/CFR-2011-title42-vol4-sec431-10.pdf"
            target="_blank"
          >
            2011
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2010-title42-vol4/pdf/CFR-2010-title42-vol4-sec431-10.pdf"
            target="_blank"
          >
            2010
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2009-title42-vol4/pdf/CFR-2009-title42-vol4-sec431-10.pdf"
            target="_blank"
          >
            2009
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2008-title42-vol4/pdf/CFR-2008-title42-vol4-sec431-10.pdf"
            target="_blank"
          >
            2008
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2007-title42-vol4/pdf/CFR-2007-title42-vol4-sec431-10.pdf"
            target="_blank"
          >
            2007
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2006-title42-vol4/pdf/CFR-2006-title42-vol4-sec431-10.pdf"
            target="_blank"
          >
            2006
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2005-title42-vol3/pdf/CFR-2005-title42-vol3-sec431-10.pdf"
            target="_blank"
          >
            2005
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2004-title42-vol3/pdf/CFR-2004-title42-vol3-sec431-10.pdf"
            target="_blank"
          >
            2004
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2003-title42-vol3/pdf/CFR-2003-title42-vol3-sec431-10.pdf"
            target="_blank"
          >
            2003
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2002-title42-vol3/pdf/CFR-2002-title42-vol3-sec431-10.pdf"
            target="_blank"
          >
            2002
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2001-title42-vol3/pdf/CFR-2001-title42-vol3-sec431-10.pdf"
            target="_blank"
          >
            2001
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-2000-title42-vol3/pdf/CFR-2000-title42-vol3-sec431-10.pdf"
            target="_blank"
          >
            2000
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-1999-title42-vol3/pdf/CFR-1999-title42-vol3-sec431-10.pdf"
            target="_blank"
          >
            1999
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-1998-title42-vol3/pdf/CFR-1998-title42-vol3-sec431-10.pdf"
            target="_blank"
          >
            1998
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-1997-title42-vol3/pdf/CFR-1997-title42-vol3-sec431-10.pdf"
            target="_blank"
          >
            1997
          </a>
          <a
            class="external"
            href="https://www.govinfo.gov/content/pkg/CFR-1996-title42-vol3/pdf/CFR-1996-title42-vol3-sec431-10.pdf"
            target="_blank"
          >
            1996
          </a>
        </div>
      </div>
      <div
        class="gov-info-source"
      >
         Source: CFR Annual Edition from 
        <a
          class="external"
          href="https://www.govinfo.gov/app/collection/cfr"
          target="_blank"
        >
           GovInfo
        </a>
         (1996â€“Present) 
      </div>
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "isUnmounted": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "unmount": [Function],
  "updateProps": [Function],
}
`;
